src/components/expanded-state/custom-gas/FeesPanel.js

```diff
@@ -0,0 +1,509 @@
+  color: colors.yellowFavorite,
+  lineHeight: 'normal',
+  size: 'smedium',
+  weight: 'heavy',
+}))`
+  ${margin(-20, 0, 20, 0)};
+`;
+
+const PanelError = styled(Text).attrs(({ theme: { colors } }) => ({
+  color: colors.red,
+  lineHeight: 'normal',
+  size: 'smedium',
+  weight: 'heavy',
+}))`
+  ${margin(-20, 0, 20, 0)}
+`;
+
+const GasTrendHeader = styled(Text).attrs(({ theme: { colors }, color }) => ({
+  color: color || colors.appleBlue,
+  size: 'smedium',
+  weight: 'heavy',
+}))`
+  ${margin(0, 5, 0, 0)};
+`;
+
+const PanelColumn = styled(Column).attrs(() => ({
+  justify: 'center',
+}))``;
+
+const Label = styled(Text).attrs(({ size }) => ({
+  size: size || 'lmedium',
+}))`
+  ${({ weight }) => fontWithWidth(weight || fonts.weight.semibold)}
+`;
+
+const GAS_FEE_INCREMENT = 1;
+const MAX_BASE_FEE_RANGE = [1, 3];
+const MINER_TIP_RANGE = [1, 2];
+
+export default function FeesPanel({
+  currentGasTrend,
+  colorForAsset,
+  onCustomGasFocus,
+}) {
+  const {
+    selectedGasFee,
+    currentBlockParams,
+    gasFeeParamsBySpeed,
+    updateToCustomGasFee,
+    updateGasFeeOption,
+    selectedGasFeeOption,
+  } = useGas();
+  const { navigate } = useNavigation();
+  const { isDarkMode } = useTheme();
+
+  const [customMaxPriorityFee, setCustomMaxPriorityFee] = useState(
+    get(selectedGasFee, 'gasFeeParams.maxPriorityFeePerGas.gwei', 0)
+  );
+  const [customMaxBaseFee, setCustomMaxBaseFee] = useState(
+    get(selectedGasFee, 'gasFeeParams.maxFeePerGas.gwei', 0)
+  );
+  const [maxPriorityFeeWarning, setMaxPriorityFeeWarning] = useState(null);
+  const [maxPriorityFeeError, setMaxPriorityFeeError] = useState(null);
+
+  const [maxBaseFeeWarning, setMaxBaseFeeWarning] = useState(null);
+  const [maxBaseFeeError, setMaxBaseFeeError] = useState(null);
+  const [feesGweiInputFocused, setFeesGweiInputFocused] = useState(false);
+
+  const maxBaseWarningsStyle = useAnimatedStyle(() => {
+    const display = !!maxBaseFeeError || !!maxBaseFeeWarning;
+    return {
+      transform: [{ scale: display ? 1 : 0 }],
+    };
+  });
+
+  const maxPriorityWarningsStyle = useAnimatedStyle(() => {
+    const display = !!maxPriorityFeeError || !!maxPriorityFeeWarning;
+    return {
+      transform: [{ scale: display ? 1 : 0 }],
+    };
+  });
+
+  const selectedOptionIsCustom = useMemo(
+    () => selectedGasFee?.option === gasUtils.CUSTOM,
+    [selectedGasFee?.option]
+  );
+
+  const { maxFee, currentBaseFee, maxBaseFee, maxPriorityFee } = useMemo(() => {
+    const maxFee = get(selectedGasFee, 'gasFee.maxFee.native.value.display', 0);
+
+    const currentBaseFee = get(currentBlockParams, 'baseFeePerGas.gwei', 0);
+    let maxBaseFee;
+    if (selectedOptionIsCustom) {
+      // block more than 2 decimals on gwei value
+      const decimals = Number(customMaxBaseFee) % 1;
+      maxBaseFee =
+        `${decimals}`.length > 4
+          ? parseInt(customMaxBaseFee, 10)
+          : customMaxBaseFee;
+    } else {
+      maxBaseFee = parseInt(
```

Comment:
could replace lines 139-152 with something like:
``` 
const selectedMaxFeePerGas = selectedGasFee?.gasFeeParams?.maxFeePerGas?.gwei ?? 0;
const maxBaseFee = toFixedDecimals(selectedOptionIsCustom ? customMaxBaseFee : selectedMaxFeePerGas, 4)
```

Reply:
or with the last argument being 2 if we want to cap it at 2 decimals per the comment

Reply:
related to https://github.com/rainbow-me/rainbow/pull/2422#discussion_r754793053

Reply:
conclusion: use code snippet above but with toFixedDecimals param being 0